File: 백준/Silver/10828. 스택/스택.py
import sys

stack = []

def main():
  n = int(sys.stdin.readline())
  results = []
  
  for _ in range(n):
    command = sys.stdin.readline().strip()
    if command.startswith("push"):
      _, x = command.split()
      stack.append(int(x))
    elif command == "pop":
      results.append(stack.pop() if stack else -1)
    elif command == "size":
      results.append(len(stack))
    elif command == "empty":
      results.append(0 if stack else 1)
    elif command == "top":
      results.append(stack[-1] if stack else -1)

  sys.stdout.write("\n".join(map(str, results)) + "\n")

if __name__ == "__main__":
  main()

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        Stack<Integer> stack = new Stack<>();
        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; i++) {
            String[] input = br.readLine().split(" ");
            switch (input[0]) {
                case "push":
                    int value = Integer.parseInt(input[1]);
                    stack.push(value);
                    break;
                case "pop":
                    if (!stack.isEmpty()) {
                        sb.append(stack.pop()).append("\n");
                    } else {
                        sb.append("-1\n");
                    }
                    break;
                case "size":
                    sb.append(stack.size()).append("\n");
                    break;
                case "empty":
                    sb.append(stack.isEmpty() ? "1\n" : "0\n");
                    break;
                case "top":
                    if (!stack.isEmpty()) {
                        sb.append(stack.peek()).append("\n");
                    } else {
                        sb.append("-1\n");
                    }
                    break;
            }
        }
        System.out.print(sb);
    }
}